sympy/functions/special/beta_functions.py

```diff
@@ -88,6 +97,11 @@ class beta(Function):
     nargs = 2
     unbranched = True
 
+    def __new__(cls, x, y=None):
+        if y == None:
+            y = x
+        return super(cls, cls).__new__(cls, x, y)
```

Comment:
Function subclasses should define `eval` rather than `__new__`.

Reply:
Yes, but how would I make sure that the no. of args are 2 when suppose I try `beta(x)`? Or should I define another class for central beta? It would be a lot of duplicate code

Reply:
```
def eval(cls, a, b=None):
    if b is None:
         return beta(a, a)
```

Reply:
This was the first thing that came to my mind. However, it does not work as `nargs=2`.
``` python
from sympy import *
>>> x=symbols('x')
>>> beta(x,x)
beta(x, x)
>>> beta(x)
Traceback (most recent call last):
  File "/home/naveen/dev/sympy/sympy/core/cache.py", line 72, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/naveen/dev/sympy/sympy/core/function.py", line 465, in __new__
    raise TypeError(temp % {
TypeError: beta takes exactly 2 arguments (1 given)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/naveen/dev/sympy/sympy/core/cache.py", line 74, in wrapper
    retval = func(*args, **kwargs)
  File "/home/naveen/dev/sympy/sympy/core/function.py", line 465, in __new__
    raise TypeError(temp % {
TypeError: beta takes exactly 2 arguments (1 given)
```
Setting `nargs=1` doesn't work either since we can't pass in 2 arguments then. This is the reason I overloaded the constructor.

Reply:
I think `nargs` can be a set like `{1, 2}`

Reply:
I just tried it and it raises an error : `ValueError: {1, 2} is not an integer`

Reply:
This is used by `log`:
```
In [1]: log.nargs
Out[1]: {1, 2}
```

Reply:
It works if I don't specify `nargs=2`